integration_tests/src/main/python/conftest.py

```diff
@@ -233,12 +234,14 @@ def std_input_path(request):
         yield path
 
 @pytest.fixture
-def spark_tmp_path(request):
+def spark_tmp_path(request, worker_id):
     debug = request.config.getoption('debug_tmp_path')
     ret = request.config.getoption('tmp_path')
     if ret is None:
         ret = '/tmp/pyspark_tests/'
-    ret = ret + '/' + str(random.randint(0, 1000000)) + '/'
+    pid = os.getpid()
+    hostname = os.uname()[1]
+    ret = f'{ret}/{hostname}-{worker_id}-{pid}-{random.randrange(0, 1<<31)}/'
```

Comment:
Can you check what happens if we run it with TEST_PARALLEL=0 without xdist?

Reply:
I manually verified this before submitting the change, uses `master` as the worker ID so there's still an identifier there.